dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp

```diff
@@ -180,28 +173,22 @@ std::optional<LocalAdmissionController::AcquireTokenInfo> LocalAdmissionControll
         if (resource_group->burstable)
             return;
 
-        // To avoid periodically_token_fetch after low_token_fetch immediately
-        if (is_periodically_fetch && !resource_group->needFetchTokenPeridically(now, DEFAULT_FETCH_GAC_INTERVAL))
+        // To avoid periodically_token_fetch after low_token_fetch immediately.
+        if (is_periodically_fetch && !resource_group->needFetchToken(now, DEFAULT_FETCH_GAC_INTERVAL))
             return;
 
-        // During trickle mode, no need to fetch tokens from GAC.
-        if (resource_group->inTrickleModeLease(now))
+        // To avoid low token fetch too freqeuent.
+        if (!is_periodically_fetch
+            && !resource_group->needFetchToken(
+                now,
+                std::chrono::duration_cast<std::chrono::seconds>(DEFAULT_LOW_TOKEN_FETCH_INTERVAL)))
```

Comment:
Maybe above two `if` can be merged into one as define `needFetchToken` as `needFetchToken(const SteadyClock::time_point & now, bool is_periodically_fetch)` and choose to use `DEFAULT_LOW_TOKEN_FETCH_INTERVAL` or `DEFAULT_FETCH_GAC_INTERVAL` inside `needFetchToken` based on `is_periodically_fetch`?

Reply:
deleted. maybe no big deal.